distribution/lib/Standard/Database/0.0.0-dev/src/Data/Take_Drop_Helpers.enso

```diff
@@ -0,0 +1,79 @@
+from Standard.Base import all
+
+import Standard.Base.Data.Index_Sub_Range as Index_Sub_Range_Module
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+from Standard.Base.Data.Index_Sub_Range import normalize_ranges, resolve_ranges, sort_and_merge_ranges
+
+from Standard.Table import Set_Mode
+
+import project.Data.Column.Column
+import project.Data.Table.Table
+from project.Errors import Unsupported_Database_Operation
+
+type Take_Drop
+    Take
+    Drop
+
+## PRIVATE
+   Apply `take` or `drop` to a table, returning the specified by the selector.
+take_drop_helper : Take_Drop -> Table -> (Index_Sub_Range | Range | Integer) -> Table
+take_drop_helper take_drop table selector =
+    check_supported selector <|
+        length = table.row_count
+        ranges = cleanup_ranges (collect_ranges take_drop length selector)
+
+        if ranges.is_empty then table.remove_all_rows else
+            # Filter on row column. Add the row column at the start, remove it at the end.
+            row_column_name = Table.make_temp_column_name
+            table_with_row_number = table.add_row_number name=row_column_name from=0
+
+            subqueries = ranges.map range->
+                generate_subquery table_with_row_number row_column_name range
+            combined = subqueries.reduce (a-> b-> a.union b)
+            combined.remove_columns row_column_name
+
+## PRIVATE
+   Turn the selector into a vector of ranges
+collect_ranges : Take_Drop -> Integer -> (Index_Sub_Range | Range | Integer) -> Vector Range
+collect_ranges take_drop length selector =
+    at x = [Range.new x x+1]
```

Comment:
This implementation of `at` is violating the contract of `take_helper`. It works because it is unused anyway.

But IMO we should rather do:
```suggestion
    at x = Panic.throw (Illegal_State.Error "Impossible: at called in Database take/drop. This is a bug in the Database library.")
```

Reply:
Done